client/components/select-dropdown/item.jsx

```diff
@@ -35,17 +36,23 @@ class SelectDropdownItem extends Component {
 			'has-icon': this.props.icon,
 		} );
 
+		const label = this.props.value || this.props.children;
+		const ariaLabel =
+			this.props.ariaLabel ||
+			( 'number' === typeof this.props.count ? `${ label } (${ this.props.count })` : label );
+
 		return (
 			<li className="select-dropdown__option">
 				<a
 					ref={ this.linkRef }
 					href={ this.props.path }
 					className={ optionClassName }
 					onClick={ this.props.disabled ? null : this.props.onClick }
-					data-bold-text={ this.props.value || this.props.children }
+					data-bold-text={ label }
 					role="menuitem"
 					tabIndex="0"
 					aria-current={ this.props.selected }
+					aria-label={ ariaLabel }
```

Comment:
The issue with option text that is read by the screen reader two times is caused by the CSS code that prepends element with its text for styling purposes:

```
	&::before {
		content: attr(data-bold-text);
    }
```

I'm not trying to remove or fix the hack, but I chose to define `aria-label` instead. It will give a developer more freedom on how the option should be read.